File: .php/Resident.php
<?php

class Resident extends User {

    private $request_history = array();
    private $request_pending = array();
    private $system;
    private $appointments = array();
    private $laundromat;
    private $requestAssigned = false;
    private $Tower;

    public function __construct($fname, $lname, $email, $Tower) {
        parent::__construct($fname, $lname, $email, "Resident");
        $this->Tower = $Tower;
    }

    public function getTower() {
        return $this->Tower;
    }

    public function getStringTower() {
        return "Tower: " . $this->Tower;
    }

    public function makeRequest($request) {
        $this->request_pending[] = $request;
        $this->system->assignRequestToAdmin();
        $this->requestAssigned = true;
    }

    public function cancelRequest($request) {
        $key = array_search($request, $this->request_pending);
        if ($key !== false) {
            unset($this->request_pending[$key]);
        }
    }

    public function viewRequestHistory() {
        foreach ($this->request_history as $request) {
            echo $request . "\n";
        }
    }

    public function viewPendingRequests() {
        foreach ($this->request_pending as $request) {
            echo $request . "\n";
        }
    }

    public function makeAppointment($appointment) {
        $this->appointments[] = $appointment;
    }

    public function viewAppointments() {
        foreach ($this->appointments as $appointment) {
            echo $appointment . "\n";
        }
    }

    public function viewAvailableTimes() {
        foreach ($this->laundromat->get_hoursOfOperation() as $time) {
            echo $time . "\n";
        }
    }

    public function viewAvailableDays() {
        foreach ($this->laundromat->get_daysOfOperation() as $day) {
            echo $day . "\n";
        }
    }

    public function cancelAppointment($appointment) {
        $key = array_search($appointment, $this->appointments);
        if ($key !== false) {
            unset($this->appointments[$key]);
        }
    }

    public function isRequestAssigned() {
        return $this->requestAssigned;
    }
}

?>

File: Software Engineering/ELR Towers Class Structure/Resident.java
import java.util.ArrayList;

public class Resident extends User{
    ArrayList<Request> request_history = new ArrayList<Request>();
    ArrayList<Request> request_pending = new ArrayList<Request>();
    ELRTowersDomesticAffairsSystem system;
    ArrayList<Appointment> appointments = new ArrayList<Appointment>();
    Laundromat laundromat;
    boolean requestAssigned = false;
    int Tower;

    public Resident (String fname,String lname, String email, int Tower){
        super(fname, lname, email, "Resident");
        
    }
    public int getTower(){
        return Tower;
    }
    public String getStringTower(){
        return "Tower: " + Tower;
    }
    public void makeRequest(Request request){
        request_pending.add(request);
        system.assignRequestToAdmin();
        requestAssigned = true;
    }
    public void cancelRequest(Request request){
        request_pending.remove(request);
    }
    public void viewRequestHistory(){
        for (Request request : request_history){
            System.out.println(request);
        }
    }
    public void viewPendingRequests(){
        for (Request request : request_pending){
            System.out.println(request);
        }
}
public void make_appointment(Appointment appointment){
    appointments.add(appointment);
}
public void view_appointments(){
    for (Appointment appointment : appointments){
        System.out.println(appointment);
    }
}
public void view_available_times(){
    for (String time : laundromat.get_hoursOfOperation()){
        System.out.println(time);
    }
}
public void view_available_days(){
    for (String day : laundromat.get_daysOfOperation()){
        System.out.println(day);
    }
}
public void cancel_appointment(Appointment appointment){
    appointments.remove(appointment);
}

public boolean isRequestAssigned() {
    return requestAssigned;
}
}

